include/tvm/ir/attrs.h

```diff
@@ -393,7 +392,7 @@ struct AttrInitEntry {
     value_missing_ = false;
     return *this;
   }
-  TSelf& describe(DMLC_ATTRIBUTE_UNUSED const char* str) { return *this; }
+  TSelf& describe(__attribute__((unused)) const char* str) { return *this; }
```

Comment:
Thanks @tkonolige for the intitial POC. Given the amount of refactoring needed, which we did not anticipate in the beginning. Let us discuss a bit more about possible alternatives a good resolution before proceeding with this PR.

A generic marco is needed in most of these cases for cross platform reasons(MSVC support